### PR TITLE
Bumping buf to v2 to build correctly and removing out of date or unused dependencies

### PR DESCRIPTION
**Pull request by andream16**
- Correctly migrating buf to v2
- Remove unused dependencies (hence why the buf.lock is clean)
- Run buf from docker for predictable outcomes
- Bumping go proto dependencies

Done in preparation of generating oscf types.

### CODE REVIEW

**Review by ptzianos — APPROVED** [review]

**Review by northdpole — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -122,7 +123,6 @@ install-lint-tools:
 	@go install github.com/kisielk/errcheck@latest
 	@go install github.com/rhysd/actionlint/cmd/actionlint@latest
 	@go install github.com/client9/misspell/cmd/misspell@latest
-	@go install github.com/bufbuild/buf/cmd/buf@v1.28.1
```
Comment (andream16) [inline Makefile:125]: Not needed since we're using docker.

File: Makefile [hunk 2]
```diff
@@ -315,9 +311,35 @@ dev-deploy: deploy-cluster dev-infra dev-dracon
 dev-teardown:
 	@kind delete clusters dracon-demo
 
-generate-protos: install-lint-tools
-	@echo "Generating Protos"
-	@buf generate
+build_buf_container:
+	$(DOCKER) build . -t $(BUF_CONTAINER) -f containers/Dockerfile.buf
+
+run_buf: build_buf_container
```
Comment (andream16) [inline Makefile:317]: Runs buf from docker for every useful command.

File: buf.yaml [hunk 3]
```diff
@@ -1,20 +1,12 @@
 ---
-version: v1
+version: v2
 lint:
   use:
-    - DEFAULT
```
Comment (andream16) [inline buf.yaml:5]: Replaced by STANDARD in buf v2

File: buf.yaml [hunk 4]
```diff
@@ -1,20 +1,12 @@
...
+    - STANDARD
     - COMMENTS
     - UNARY_RPC
     - PACKAGE_NO_IMPORT_CYCLE
   except:
     # we use Please to run protoc where we only need the last part of the
     # package name to match the directory name.
     - PACKAGE_DIRECTORY_MATCH
-
-deps:
```
Comment (andream16) [inline buf.yaml:14]: These were unused.

File: containers/Dockerfile.buf [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# Setup Go alpine to install protoc-gen-go plugin.
+FROM golang:alpine AS golang
+ENV GO111MODULE=on
+RUN go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.35.1
```
Comment (andream16) [inline containers/Dockerfile.buf:4]: The buf image doesn't come with ways to configure plugins so this is what was recommended by their team on various Github issues.

File: Makefile [hunk 6]
```diff
@@ -315,9 +311,35 @@ dev-deploy: deploy-cluster dev-infra dev-dracon
 dev-teardown:
 	@kind delete clusters dracon-demo
 
-generate-protos: install-lint-tools
-	@echo "Generating Protos"
-	@buf generate
+build_buf_container:
```
Comment (dlicheva) [inline Makefile:314]: Tiny code style one: all other `make` commands use kebab case. Could you change that to match them please?

**Review by ptzianos — APPROVED** [review]

**Review by dlicheva — APPROVED** [review]
Thank you :)

**Review by dlicheva — APPROVED** [review]

**Pull request merged**